packages/core/src/messages/responses/vector/vector-get-item-metadata-batch.ts

```diff
@@ -0,0 +1,84 @@
+import {SdkError} from '../../../errors';
+import {ResponseBase, ResponseError, ResponseSuccess} from '../response-base';
+
+/**
+ * Parent response type for a VectorGetItemMetadataBatch request. The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Success}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof VectorGetItemMetadataBatch.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `VectorGetItemMetadataBatch.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {
+  hits():
+    | Record<string, Record<string, string | number | boolean | Array<string>>>
```

Comment:
It's becoming tedious to copy the metadata `Record` type signature. I prefer extracting this to a type alias in a single place. If we did this though then the type would not expand in IDE tooltips; users would have to go to the method definition then hover over the metadata alias to understand what it was.

Though that's less than ideal for users, the duplication is becoming a bit much; should we add a new metadata type alternative in the future, we will have to update it in many places. Thus I recommend extracting the type

Reply:
agree

Reply:
Refactored in [025f3c0](https://github.com/momentohq/client-sdk-javascript/pull/1058/commits/025f3c0387cf541f89395942e50db92cf3307247)